pkg/streamingpromql/operators/functions/range_vectors.go

```diff
@@ -483,11 +483,14 @@ var Deriv = FunctionOverRangeVectorDefinition{
 	NeedsSeriesNamesForAnnotations: true,
 }
 
-func deriv(step *types.RangeVectorStepData, _ float64, emitAnnotation types.EmitAnnotationFunc) (float64, bool, *histogram.FloatHistogram, error) {
+func deriv(step *types.RangeVectorStepData, _ float64, _ []types.ScalarData, _ types.QueryTimeRange, emitAnnotation types.EmitAnnotationFunc) (float64, bool, *histogram.FloatHistogram, error) {
 	fHead, fTail := step.Floats.UnsafePoints()
 
-	if len(fHead)+len(fTail) == 1 && step.Histograms.Any() {
+	if step.Floats.Any() && step.Histograms.Any() {
 		emitAnnotation(annotations.NewHistogramIgnoredInMixedRangeInfo)
+	}
+
+	if len(fHead)+len(fTail) == 1 && step.Histograms.Any() {
```

Comment:
I don't think we need this condition any more. The annotation will be emitted, and then if there is less than 2 points the result is the same.

Reply:
Good catch, fixed in 042c4ab